.ci/Jenkinsfile

```diff
@@ -124,9 +124,16 @@ pipeline {
       post {
         always {
           archiveArtifacts(allowEmptyArchive: true, artifacts: '${BASE_DIR}/**/mail-out-test*.html')
-          junit(allowEmptyResults: true,
-            keepLongStdio: true,
-            testResults: "${BASE_DIR}/target/surefire-reports/junit-*.xml,${BASE_DIR}/target/surefire-reports/TEST-*.xml")
+          withEnv([
+            "JUNIT_2_OTLP=true",
+            "OTEL_SERVICE_NAME=apm-pipeline-library",
+            "JUNIT_OTEL_SERVICE_VERSION=main",
+            "JUNIT_OTEL_TRACE_NAME=junit-tests"
```

Comment:
You can simplify this with:

```groovy
pipeline {
  agent { label 'linux && immutable' }
  environment {
    JUNIT_2_OTLP = "true"
    OTEL_SERVICE_NAME = "apm-pipeline-library"
    JUNIT_OTEL_SERVICE_VERSION = "main"
    JUNIT_OTEL_TRACE_NAME = "junit-tests"
```

or even at the stage level, then we don't need to change the junit call with a closure:

```groovy
stage('Test') {
  environment {
    JUNIT_2_OTLP = "true"
    OTEL_SERVICE_NAME = "apm-pipeline-library"
    JUNIT_OTEL_SERVICE_VERSION = "main"
    JUNIT_OTEL_TRACE_NAME = "junit-tests"
  }
  steps {
```

Reply:
I wanted to make it explicit, although I'm fine with enabling them at the pipeline level

Reply:
I added this support, updating the PR description